[abseil-cpp] Remove unncessary <iostream> includes

Submitted upstream as https://github.com/abseil/abseil-cpp/pull/1801
but Google requested to implement this remaining part of the patch in
a different way ("change that code to output to a std::string and let
the user dump it").

If we don't see them fix it the next few months, maybe we can try re-
submitting a new patch using that other approach, ourselves.

--- absl/strings/cord.cc
+++ absl/strings/cord.cc
@@ -23,7 +23,6 @@
 #include <cstring>
 #include <iomanip>
 #include <ios>
-#include <iostream>
 #include <limits>
 #include <memory>
 #include <ostream>

--- absl/strings/internal/cord_rep_btree.cc
+++ absl/strings/internal/cord_rep_btree.cc
@@ -17,10 +17,13 @@
 #include <atomic>
 #include <cassert>
 #include <cstdint>
-#include <iostream>
 #include <ostream>
 #include <string>
 
+#ifndef NDEBUG
+#include <iostream>
+#endif
+
 #include "absl/base/attributes.h"
 #include "absl/base/config.h"
 #include "absl/base/internal/raw_logging.h"

--- absl/strings/internal/cord_rep_btree_navigator.h
+++ absl/strings/internal/cord_rep_btree_navigator.h
@@ -16,7 +16,6 @@
 #define ABSL_STRINGS_INTERNAL_CORD_REP_BTREE_NAVIGATOR_H_
 
 #include <cassert>
-#include <iostream>
 
 #include "absl/strings/internal/cord_internal.h"
 #include "absl/strings/internal/cord_rep_btree.h"
